core/geometry/src/test/curve/CurveCurveIntersectXY.test.ts

```diff
@@ -2356,11 +2361,221 @@ describe("CurveCurveIntersectXY", () => {
+    numExpected: number, dx: number, dy: number,
+  ) {
+    GeometryCoreTestIO.captureCloneGeometry(allGeometry, curve0, dx, dy);
+    GeometryCoreTestIO.captureCloneGeometry(allGeometry, curve1, dx, dy);
+
+    const testSpiralIntersection = (intersections: CurveLocationDetailPair[], reversed: boolean) => {
+      GeometryCoreTestIO.captureCurveLocationDetails(allGeometry, intersections, 5, dx, dy);
+      const curveName0 = curve0.constructor.name;
+      const curveName1 = curve1.constructor.name;
+      ck.testCoordinate(
+        numExpected,
+        intersections.length,
+        `expect ${numExpected} intersection(s) between ${curveName1} and ${curveName0}`,
+      );
+      for (const intersection of intersections) {
+        let curve0Detail = intersection.detailA;
+        let curve1Detail = intersection.detailB;
+        const oneCurveIsSpiral = curve0Detail.curve instanceof TransitionSpiral3d || curve1Detail.curve instanceof TransitionSpiral3d;
+        const bothChain = curve0 instanceof CurveChainWithDistanceIndex && curve1 instanceof CurveChainWithDistanceIndex;
+        const oneChainOneCollection = (curve0 instanceof CurveChainWithDistanceIndex && curve1 instanceof CurveCollection)
+          || (curve1 instanceof CurveChainWithDistanceIndex && curve0 instanceof CurveCollection);
+        if (reversed && (oneCurveIsSpiral || bothChain || oneChainOneCollection)) {
+          curve0Detail = intersection.detailB;
+          curve1Detail = intersection.detailA;
+        }
+        if (!(curve0 instanceof CurveCollection)) {
+          const pointOnCurve0 = curve0Detail.point;
+          const intersectionPoint0 = curve0.fractionToPoint(curve0Detail.fraction);
+          ck.testPoint3dXY(
+            pointOnCurve0,
+            intersectionPoint0,
+            `intersection point on ${curveName0} should match ${curveName0}.fractionToPoint`,
+          );
+        }
+        if (!(curve1 instanceof CurveCollection)) {
+          const pointOnCurve1 = curve1Detail.point;
+          const intersectionPoint1 = curve1.fractionToPoint(curve1Detail.fraction);
+          ck.testPoint3dXY(
+            pointOnCurve1,
+            intersectionPoint1,
+            `intersection point on ${curveName1} should match ${curveName1}.fractionToPoint`,
+          );
+        }
+      }
+    }
+    // test both paths
+    const intersectionsAB = CurveCurve.intersectionXYPairs(curve0, false, curve1, false);
+    testSpiralIntersection(intersectionsAB, false);
+    const intersectionsBA = CurveCurve.intersectionXYPairs(curve1, false, curve0, false);
+    testSpiralIntersection(intersectionsBA, true);
+  };
+
+  it("SpiralIntersection", () => {
+    const ck = new Checker();
+    const allGeometry: GeometryQuery[] = [];
+    let dx = 0;
+    let dy = 0;
+
+    const rotationTransform = Transform.createFixedPointAndMatrix(
+      Point3d.create(70, 0),
+      Matrix3d.createRotationAroundVector(Vector3d.create(0, 0, 1), Angle.createDegrees(90))!,
+    );
```

Comment:
Can you also test intersections between curves that are _not_ coplanar?  The true superpower of `CurveCurve.intersectionXYPairs` is that you can throw any two fully 3D curves into it, and it will return their virtual intersections as seen from the top view (non-camera). This is useful e.g., for snapping to space curves at virtual intersections in any view.